.github/workflows/cd.yml

```diff
@@ -50,11 +50,25 @@ jobs:
 
     - name: Upload Artifacts to S3
       run: |
-        s3_path=s3://artifacts.opendistroforelasticsearch.amazon.com/downloads
-        aws s3 cp artifacts/*.zip $s3_path/elasticsearch-plugins/opendistro-security/
-        aws s3 cp artifacts/*.deb $s3_path/debs/opendistro-security/
-        aws s3 cp artifacts/*.rpm $s3_path/rpms/opendistro-security/
-        aws cloudfront create-invalidation --distribution-id ${{ secrets.DISTRIBUTION_ID }} --paths '/downloads/*'
+        zip=`ls artifacts/*.zip`
+        rpm=`ls artifacts/*.rpm`
+        deb=`ls artifacts/*.deb`
+
+        # Inject the build number before the suffix
+        zip_outfile=`basename ${zip%.zip}-build-${GITHUB_RUN_NUMBER}.zip`
```

Comment:
Will it be better to modify `pom.xml` and `build.gradle` to produce output files in the required format by passing `${GITHUB_RUN_NUMBER}`?

Reply:
I would not modify those, because it would also change the names of the artifacts that go into Sonatype and the Github release. I think those artifacts do not want the build number.

Reply:
Instead of github build number, can we use commit sha? This will allow us to build the same artifact if necessary.

Reply:
That's an option, but it makes it harder for us to know what the latest build is. The GITHUB_RUN_NUMBER increases monotonically, which makes figuring out what to promote to production easier. If you feel strongly then I'm happy to change it - the release process doesn't rely on human-readable names (or, at least, won't rely on those names soon).

Reply:
This will give us an option to trigger CD workflow without impacting infra build for example when we tag release candidate commit with the final release tag.

Reply:
The GITHUB_RUN_NUMBER also ensures that you will not impact the infra build.

Reply:
In the interests of getting this unblocked, would you be OK with shipping this change as-is, and if we need to come back and change it to a hash then we can do that after the next release? This isn't a one-way door.

Reply:
It is not so much important for me whether it is commit hash or github build number as long as I can re-run CD workflow on the same commit without impacting/confusing infra build. In case github build number will be incremented each time CD build runs, will not that cause confusion whether it is the same artifact as already published or a new artifact?

Reply:
That's OK, it will not cause any confusion. Once we select, say, build-15 as the version that goes into the next ODFE then that is locked in. If build-16 is identical then that's OK - we don't care. Once the release train has departed then there are no more changes :)